sdk/core/core/src/az_auth.c

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_auth.h>
+
+#include <az_contract.h>
+#include <az_http_client.h>
+#include <az_http_request_builder.h>
+#include <az_json_read.h>
+#include <az_str.h>
+#include <az_uri.h>
+
+#include <assert.h>
+
+#include <_az_cfg.h>
+
+AZ_NODISCARD az_result az_auth_init_client_credentials(
+    az_auth_credentials * const out_result,
+    az_span const tenant_id,
+    az_span const client_id,
+    az_span const client_secret) {
+  AZ_CONTRACT_ARG_NOT_NULL(out_result);
+  AZ_CONTRACT_ARG_VALID_SPAN(tenant_id);
+  AZ_CONTRACT_ARG_VALID_SPAN(client_id);
+  AZ_CONTRACT_ARG_VALID_SPAN(client_secret);
+
+  *out_result = (az_auth_credentials){
+    .tenant_id = tenant_id,
+    .kind = AZ_AUTH_KIND_CLIENT_CREDENTIALS,
+    .data = { .client_credentials = { .client_id = client_id, .client_secret = client_secret } }
+  };
+
+  return AZ_OK;
+}
+
+enum {
+  MIN_BUFFER = 1350, // if you measure the length of the login.microsoftonline.com's response, it is
```

Comment:
IMHO, we should always use `az_` prefix even for internal things.

Reply:
Would be nice to distinguish the internal from external with "_az_...."

Reply:
Underscored + starting with capital letter names are reserved. Let's discuss later, what we can do. I'll use AZ_ for now.
I am also not sure that I agree that we have to use az_prefixes for internal stuff that is only visible within a single unit of compilation (statics inside a .c file). But it definitely won't hurt to use AZ_ for now. Let's discuss later.